dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -5,7 +5,15 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
   # GET /users/auth/:provider/callback
   def all
-    @user = User.from_omniauth(request.env["omniauth.auth"], request.env['omniauth.params'])
+    auth_hash = request.env['omniauth.auth']
+    auth_params = request.env['omniauth.params']
+
+    # Fiddle with data if it's a Powerschool request (other OpenID 2.0 providers might need similar treatment if we add any)
+    if request.env["omniauth.auth"].provider.to_s == 'powerschool'
+      auth_hash = extract_powerschool_data(request.env["omniauth.auth"])
+    end
```

Comment:
Style: It might be nice for `auth_hash` to become a private method on this controller, and the only place we reference `request.env['omniauth.auth']`.  Then it'd be a fairly obvious place to add support for other OpenID 2.0 providers in the future.